mailpoet/mailpoet
# Add new "used shipping method" segment [MAILPOET-4992]

## Description

This PR adds a new WooCommerce segment called "used shipping method".

## Code review notes

I based this new segment on the "used payment method" segment.

I noticed that when the operator "none of" is used for the "used payment method" it matches not only WooCommerce customers but all types of subscribers. I followed the same logic for this new segment but I wonder if we should limit the scope of both segments only to WooCommerce customers? (cc @NeosinneR)

## QA notes

How to test this PR:

- Place a few WooCommerce orders using different shipping methods.
- Go to the page to create a new MailPoet segment (/wp-admin/admin.php?page=mailpoet-segments#/new-segment) and try different values for the parameters of the new "used shipping method" segment.
- Verify that the number of matched subscribers matches what is expected.
- Send an email to this segment and verify that the recipients match what is expected.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-4992]

## After-merge notes

_N/A_


[MAILPOET-4992]: https://mailpoet.atlassian.net/browse/MAILPOET-4992?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
